src/tests/workflows/recipes/components/FilterObjectForm.test.js

```diff
@@ -0,0 +1,509 @@
+    expect(getByText('Geo (2)')).toBeTruthy();
+    // Same with "Browser (2)" because both channels and versions is chosen.
+    expect(getByText('Browser (2)')).toBeTruthy();
+    // And "Sampling (1)" because a sampling method has been chosen.
+    expect(getByText('Sampling (1)')).toBeTruthy();
+
+    expect(fetchMock.called()).toBeTruthy();
+  });
+
+  describe('<serializeFilterObjectToMap>', () => {
+    it('serialize an empty List to an empty Map', () => {
+      const map = serializeFilterObjectToMap(List());
+      expect(map).toBeImmutableMap();
+      expect(map.size).toEqual(0);
+    });
+
+    it('serialize an non-empty List to a Map', () => {
+      const list = fromJS([
+        { type: 'channel', channels: ['beta'] },
+        {
+          type: 'bucketSample',
+          start: 50,
+          count: 70,
+          total: 100,
+          input: ['normandy.request_time'],
+        },
+      ]);
+      const map = serializeFilterObjectToMap(list);
+      expect(map).toBeImmutableMap();
+      expect(map.size).toEqual(2);
+      expect(map.get('channels')).toEqual(List(['beta']));
+      expect(map.get('_sampling')).toEqual(
+        fromJS({
+          type: 'bucketSample',
+          start: 50,
+          count: 70,
+          total: 100,
+          input: ['normandy.request_time'],
+        }),
+      );
+    });
+
+    it('serialize an undefined List to an empty Map', () => {
+      const map = serializeFilterObjectToMap(undefined);
+      expect(map).toBeImmutableMap();
+      expect(map.size).toEqual(0);
+    });
+
+    it('throw error on something truthy that is not a List', () => {
+      expect(() => {
+        serializeFilterObjectToMap([]);
+      }).toThrow("'' (object) is not a List");
+    });
+    it('throw error on an unrecognized key', () => {
+      const list = fromJS([{ type: 'color', colors: ['red'] }]);
+      expect(() => {
+        serializeFilterObjectToMap(list);
+      }).toThrow("Not sure how to convert 'color'");
+    });
+  });
+
+  describe('<deserializeFilterObjectToList>', () => {
+    it('deserialize an undefined Map to an empty List', () => {
+      const map = deserializeFilterObjectToList(undefined);
+      expect(map).toBeImmutableList();
+      expect(map.size).toEqual(0);
+    });
+
+    it('serialize an non-empty Map to a List', () => {
+      const obj = {
+        channels: ['beta'],
+      };
+      const list = deserializeFilterObjectToList(obj);
+      expect(list).toBeImmutableList();
+      expect(list.size).toEqual(1);
+      expect(list.first()).toEqual(fromJS({ type: 'channel', channels: ['beta'] }));
+    });
+
+    it('throw an error on unrecognized names', () => {
+      const obj = {
+        colors: ['red'],
+      };
+      expect(() => {
+        deserializeFilterObjectToList(obj);
+      }).toThrow("Key 'colors' no in known mapping.");
+    });
+
+    it('serialize a map with exception _sampling', () => {
```

Comment:
What does `exception` mean here?

Reply:
How about `should serialize a map with the exceptional _sampling`